File: src/main/java/com/syfe/personalfinance/dto/CategoryDto.java
package com.syfe.personalfinance.dto;

import com.syfe.personalfinance.enums.CategoryType;
import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.NotNull;
import lombok.*;

import java.util.List;

public class CategoryDto {

    @Getter
    @Setter
    @NoArgsConstructor
    @AllArgsConstructor
    @Builder
    public static class CreateCategoryRequest {
        @NotBlank(message = "Category name is required")
        private String name;

        @NotNull(message = "Category type (INCOME or EXPENSE) is required")
        private CategoryType type;
    }

    @Getter
    @Setter
    @NoArgsConstructor
    @AllArgsConstructor
    @Builder
    public static class CategoryResponse {
        private String name;
        private CategoryType type;
        private boolean isCustom;
    }

    @Getter
    @Setter
    @NoArgsConstructor
    @AllArgsConstructor
    @Builder
    public static class CategoryListResponse {
        private List<CategoryResponse> categories;
    }
}

File: src/main/java/com/syfe/personalfinance/dto/CategoryDto.kt
package com.syfe.personalfinance.dto

import com.syfe.personalfinance.enums.CategoryType
import jakarta.validation.constraints.NotBlank
import jakarta.validation.constraints.NotNull

class CategoryDto {

    data class CreateCategoryRequest(
        @field:NotBlank(message = "Category name is required")
        val name: String = "",

        @field:NotNull(message = "Category type (INCOME or EXPENSE) is required")
        val type: CategoryType? = null
    )

    data class CategoryResponse(
        val name: String = "",
        val type: CategoryType? = null,
        val isCustom: Boolean = false
    )

    data class CategoryListResponse(
        val categories: List<CategoryResponse> = emptyList()
    )
}
